src/Common/tests/System/PlatformDetection.cs

```diff
@@ -34,6 +34,58 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static int s_isWinRT = -1;
+
+        public static bool IsWinRT
+        {
+            get
+            {
+                if (s_isWinRT != -1)
+                    return s_isWinRT == 1;
+
+                if (!IsWindows || IsWindows7)
+                {
+                    s_isWinRT = 0;
+                    return false;
+                }
+
+                byte[] buffer = new byte[0];
+                uint bufferSize = 0;
+                try
+                {
+                    int result = GetCurrentApplicationUserModelId(ref bufferSize, buffer);
+                    switch (result)
+                    {
+                        case 15703: // APPMODEL_ERROR_NO_APPLICATION
+                            s_isWinRT = 0;
+                            break;
+                        case 0:     // ERROR_SUCCESS
```

Comment:
We do have that, but because this is a test support file I explicitly didn't want to bring in the file dependency.